### DISCUSSION TITLE
[Feature Request] Don't depend on Flatpak when building from source

### DISCUSSION THREAD

**Discussion by yochananmarqos** [CrankshaftForSteam/crankshaft#18] [Bug Report] [answered]
This can work building from source using Steam installed via distro package managers, but the Flatpak commands spoil it for those who do not use Flatpaks.

**Comment by coolavery** [comment]
Hi, thanks for the request!

This one has been on my todo list for a while. Currently, when building/running Crankshaft, there's a dev mode, which will recompile assets and won't use any Flatpak stuff. When not running in dev mode, assets will be compiled and embedded in the output binary, and commands will be run through `flatpak-spawn`.

I basically just have to split out the Flatpak behaviour from the rest of the dev mode behaviour. I'll see if I have time to do that this week!

I don't think I'll want to officially support any other types of packages, since just testing and releasing the Flatpak already takes a lot of time, but at least this will make it easier for others to create distro packages.

btw, I don't know if I misunderstood this:

> using Steam installed via distro package managers

But if you weren't aware, the Flatpak version of Crankshaft will work with both the native (distro packaged) and Flatpak versions of Steam.

**Comment by coolavery** [comment, marked as answer]
Sorry, it took longer than I expected, but non-Flatpak builds are now fully working. The `build` target in the Makefile will build a release-mode Crankshaft binary that doesn't depend on Flatpak.

**Comment by yochananmarqos** [comment]
Thank you! It's working great and I've added it to the AUR: [`crankshaft`](https://aur.archlinux.org/packages/crankshaft)

**Comment by pastaq** [comment]
> Thank you! It's working great and I've added it to the AUR: [`crankshaft`](https://aur.archlinux.org/packages/crankshaft)

I've updated [crankshaft-bin](https://aur.archlinux.org/packages/crankshaft-bin) with your recommendations. Thanks for the tips. I also added `crankshaft` as a conflicts for `crankshaft-bin`. 

BTW, why it was taking so long... A bit of experimentation was going on.  I can't seem to get ChimeraOS to recognize the service file, and manually running crankshaft from ChimeraOS seems to error out when it attempts to load it itself as well. Any ideas what could be causing that? I don't seem to have this issue in Manjaro using my package, it seems to work fine with manual or automatic enabling. I get the same symptoms from your package as well in ChimeraOS.

**Comment by coolavery** [comment]
FYI I converted this issue into a discussion since we're closing Github issues, and there's still active discussion on this ticket.

Also, let me know if I can help with debugging or improvements to the autostart system!

**Comment by yochananmarqos** [comment]
I don't know anything about ChimeraOS, sorry.
